### DISCUSSION TITLE
STMicroelectronics STM32 Nucleo-64 MCU Development Board NUCLEO-F446RE

### DISCUSSION THREAD

**Discussion by amirpdn** [stm32duino/Arduino_Core_STM32#1386] [Q&A] [answered]
Dear Sir/Madam,
I hope you are doing well,
We are working on a project in which 50 pulses should be generated with a particular delay, and these pulses should be repeated in a period (0.02 seconds).
We are using a Nucleo-F446RE board to do the project, 
We want to know how we could use the MCU's timers to produce such pulses?
Is there any way to use Timers with a specific delay to generate 50 pulses?
I am looking forward to hearing from you,
Many thanks.
poordadashnia@gmail.com

**Comment by ABOSTM** [comment, marked as answer]
Hi @amirpdn,
You can use HadwareTimer library to generate the pulses:
https://github.com/stm32duino/wiki/wiki/HardwareTimer-library
Several examples are provided:
https://github.com/stm32duino/wiki/wiki/HardwareTimer-library#4-examples

About your 50 pulses, you can for example attach an interrupt to count the number of pulses and stop timer once 50 pulses are reached.
